--!strict

warn("[Flux/Client] Loaded CharacterClass");

local CharacterClass = {};

CharacterClass.__index = CharacterClass;
CharacterClass.__mode = "KV";

local require, network, tagged, asset, tween, thread, element = shared.import("require", "network", "tagged", "asset", "tween", "thread", "getasset");
local wait, delay, spawn, defer, cancel = task.wait, task.delay, task.spawn, task.defer, task.cancel;

local Player = game:FindService("Players").LocalPlayer;

function CharacterClass.new(Player: (Player) )
    local charMeta = setmetatable({}, CharacterClass);
  
    charMeta.Meter = 20;
    charMeta.MaxMeter = 20;
    
    charMeta.Sprinting = false;
    charMeta.ShiftHeld = false;
    
    charMeta.CanRun = true;
    
    charMeta.Blinking = false;
    charMeta.BlinkMeter = 20;
    
    charMeta.Counting = true;
	  charMeta.IsIn294 = false;
  
	  charMeta.RadioDebounce = false;
	  charMeta.IsWatching = false;
	
    return charMeta;
end;


return CharacterClass;

--              ░░   ░ ░▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▒▒▓▓▓▒ ░░      ░░░ ░░░░░░     
--           ░░░░░    ░░▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░░░░     ░    ▒▓░░ ░    
--    ░░    ░░░░░   ░ ░▒▓▓▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▒▒▒▓▓▓▓░░░░░      ░   ░▒░░     
--░░          ░░░     ░▓▒▒▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▒▒▒▒▓▓▓▓▒░▒▒░░░░    ░░ ░░░░     
--  ░░       ░░░░░ ░ ░▒▒▒▒▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▒▒▒▓▓▓▒▒▓▓▒░░░       ░░░░░    
--         ░░░░   ░  ░▓▓▒▒▓▓▓▓▓▓▓▓▓▓▓▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓░▒▒▒▒░       ░ ░░░░░   
--          ░░░░     ░▓▒▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▒░▓▓▓▓░        ░▓▒░░   
--          ░     ░ ░▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▒░▓▓▓▒░░        ▒▓▒░░  
--     ░░░     ░░   ░▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▒░▒▓▒▒░░░  ░    ░▓▓░   
--     ░░░░░░░ ░  ░ ░▒▓▓▓▓▒▒▒▒▒▒▒▒▒▒▒▒▒▒▓▒▒▒▒▒▒▒▒▒▒░░░░░░░░░░▒▓▒░          ▒▒░   
--    ░░░░░░░░  ░░░░░░░░░░░░░░░░░░░░░░░░░▒░░░░░░░            ░▓▒░░        ░▒▒░░  
--    ░░░░░ ░  ░    ░░░          ░░░░░░░░░░░░                ▒▓▒░░░░      ░▒▒░░  
--    ░░░░     ░░░░ ░░░░░░         ░░░░░░░░                  ░▓▒░░░   ░ ░  ░▓▓▒  
--    ░░░░    ░░░░░  ░░░░            ░░░░░░       ░          ░▒▒░░      ░  ░▒▓▒  
--    ░░░░    ░░░░░  ░░░░░░░░       ░░░▒▓▒░░        ░░░░░░░░ ░▒▓▒░░ ░ ░  ░░░▓▓▒  
--   ░░░░░    ░░░░░  ░░░▒░▒░       ░░▒▓▓▓▓▓▒░       ░░░░▒▒░ ░░▒▓▒░░ ░  ░   ░▓▓░  
--  ░░░░░░░  ░░░░░░  ░░░░░░░░     ░░▒▓▓▓▓▓▓▓▒░░    ░░░░░░░░░ ░▒▓▒░░ ░░      ▒▒░  
--  ░░░░░░  ░░░░░ ░░░░░░░░░░░░░░░░▒▓▓▓▓▓▓▓▓▓▓▓▒░░░░░░░░░    ░░▓▓▓░░░        ░░░  
-- ░░░░░░░ ░░░░░░░  ░░░░░░░░░░▒▒▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▒▒▒▒░░░░░░░░░░░▓▓▓░ ░░       ░▒▒  
--  ░░░░░░░░░░░░░  ░░░▒▒▒▒▒▒▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▒▒▒▒▒▒▒▒░░░░░░▒▓▓░░░░        ▒▒  
--  ░░░░░░░░░░░░░░░░░▒▒▒▒▒▓▓▓▓▓▓▒▓▒▓▓▓▓▓▓▓▓▓▓▓▓▒▒▒▒▒▓▒▒▒▒▒▒▒▒░▓▓▓▒░░░░       ▒▒  
-- ░░░░░░░░░▒░░░░░  ░▒▒▒▒▓▓▓▓▓▓▓▒▒▒▓▓▓▓▓▓▓▓▓▓▓▓▒▒▒▒▒▓▓▓▓▓▓▓▒▒░▒▓▓▒░░░░░░     ░▒  
--  ░░░░░░░░▒░░░ ░   ░▒▓▓▓▓▓▓▓▓▒░▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▒░▒▓▓▓▓▓▓▓▓▓▒░▒▓▓▒░░░░░      ░░  
--  ░░░░░░░▓▒░░      ░▒▒▓▓▓▓▓▓▓▒▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▒▒░▓▓▒░░░░░░░     ░▒  
--  ░░░░░░▒▒▒░░░   ░░░░▒▒▒▓▓▓▓▓▒▒▒░░░░░▒▒▒▒░░░░░░░░▒▓▓▓▓▒▒░░░░▒▓▓░░░░░  ░░    ░  
--  ░░░░░░▒▓▓░░ ░ ░ ░░░░▒▒▒▓▓▓▓▓░░    ░░░░░░    ░░░▓▓▓▓▒▒▒░░░░▓▓▒░░  ░░       ░  
--░░░░░░ ▒▓▓▒░░   ░░░░░░▒▒▓▓▓▓▓▓░                 ░▓▓▓▓▓▒░░░░░▓▓▒░ ░░            
--  ░░░░░▒▓▓▒░░░░░░ ░░ ░░▒▓▓▓▓▓▓▒░               ░░▒▓▓▓▓▒▒░░░░▓▓▒░░░             
--  ░░░░░▒▒▒░░ ░░    ░░ ░▓▓▓▓▓▓▓▓▒▒░░░      ░░░░░░▒▓▓▓▓▓▒▒░░░▓▓▓▓░░░  ░░░░       
-- ░░░░░░▓▓░░  ░░     ░░░▒▒▒▓▓▓▓▓▓▓▓▒▓▓▓▒▓▓▓▒▒▒▒▓▓▒▒▒▒▓▒▒▒░░░▒▓▓░ ░░   ░         
-- ░░░░░░▓▓▒░            ░▒▒▒▓▓▒░░░▒░░░▒▒▓▓▒▒░░░░░▒▓▓▒▒▒▒░░░░▒▓▓▒░░░  ░░         
-- ░░░░░▓▓▓▒░░░░░      ░░░░░░░░░░░░░░░░░░░░░░░░░░   ░░░░░░░░░▒▓▓▓░░░░  ░░        
--░░░░░░▓▓▓░░░    ░    ░░░░          ░  ░░             ░░░░ ░░▓▓▒░░░░   ░        
-- ░░░░░▓▓▓░░░  ░ ░      ░░░░  ░░░░░      ░  ░░░░░░  ░░░░   ░▒▓▓▒░░░    ░░       
--░░░░░░▓▓▓░░    ░         ░░░▒▒▓▓▓▓▓▒░░░░▒▓▓▓▓▓▓▒░░░░░     ░▒▓▓▓░░░░   ░░       
--░░░░░░▓▓▒░░   ░░░      ░░░░░░░░▓▓▓▓▓▓▓▒▓▓▓▓▓▓▒░░░░░        ░▓▓▓░░░░    ░       
--░░░░░░▓▓▓░░    ░░         ░  ░░░░░▒▒░░░░░░░░░░            ░▒▓▓▒░       ░░      
--░░░░░▓▓▓▓░░░  ░         ░░        ░░░░░░░░░░              ░░▒▓▒░       ░░      
--░░░░░▓▓▓▒░                                               ░░▒▓▓▒░       ░░░     
--░░░░░▓▓▓▓░░                                               ░▒▓▓▒░░        ░     
-- ░░░░▒▓▓▒░░                                               ░░▓▓▒░      ░░░░░    
--░░░░░▓▓▒▒░                                                ░░▓▓▒░░░░   ░░░░░░   
--░░░░░▒▓▓▒░░                                             ░ ░░▒▒▒░░     ░░░░▒░░  
